.github/workflows/docker-publish.yaml

```diff
@@ -0,0 +1,41 @@
+name: Create and publish a Docker image
+
+on:
+  push:
+    branches: ['hugo']
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          registry: ${{ env.REGISTRY }}
```

Comment:
Hi @apoorvkh, thanks for this PR! I noticed that this line mentions some new variables. What account does this use to push to the container registery?

Also, could you create a new documentation page (maybe `docker.md` or something) detailing how to set this up? Pretty much just what you had for your PR comment but as documentation :)

Reply:
You're welcome! This workflow is simply from [Publishing images to Github Packages](https://docs.github.com/en/actions/publishing-packages/publishing-docker-images#publishing-images-to-github-packages) and, following the environment variables (e.g. `${{ github.actor }}`), the user who "triggers the workflow" (in this case, by pushing to the `hugo` branch) will push to the container registry. This might just be you, as you push commits or merge PRs. And, the image will be at `ghcr.io/jackyzha0/quartz:hugo` after you merge this PR, since `IMAGE_NAME` is set to the variable `${{ github.repository }}`.

Reply:
I also added the documentation page!